server/methods/catalog.js

```diff
@@ -752,6 +756,8 @@ Meteor.methods({
         results.push(result);
       }
     }
+
+    Hooks.Events.run("onProductClone", products);
```

Comment:
`onProductClone` should probably included both the source and the destination products.